src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -390,6 +393,42 @@ private SmugglerProgressBase.Counts ProcessDocuments(SmugglerResult result, Buil
             return result.Documents;
         }
 
+        private SmugglerProgressBase.Counts ProcessLegacyAttachments(SmugglerResult result)
+        {
+            using (var actions = _destination.Documents())
+            {
+                foreach (var item in _source.GetLegacyAttachments(actions))
+                {
+                    _token.ThrowIfCancellationRequested();
+                    result.LegacyAttachments.ReadCount++;
+
+                    if (result.LegacyAttachments.ReadCount % 1000 == 0)
+                    {
+                        var message = $"Read {result.LegacyAttachments.ReadCount:#,#;;0} legacy attachments.";
+                        result.AddInfo(message);
+                        _onProgress.Invoke(result.Progress);
+                    }
+                    //TODO:return null if reading the attachment fails
```

Comment:
Why return null, and what can fail here?

Reply:
Blittable reading can throw, no?

Reply:
Not if you are reading valid data